commit 26bb2dc102783fef49336b26a94563318f9790d3
Merge: aa628ebb069e a187c1b0a800
Author: Dave Airlie <airlied@redhat.com>
Date:   Fri Oct 11 08:01:16 2024 +1000

    Merge tag 'drm-xe-next-2024-10-10' of https://gitlab.freedesktop.org/drm/xe/kernel into drm-next
    
    Cross-subsystem Changes:
    - Add drm_line_printer (Michal)
    
    Driver Changes:
    - Fix an UAF (Matt Auld)
    - Sanity check compression and coherency mode (Matt Auld)
    - Some PIC-ID work (Jani)
    - Use IS_ENABLED() instead of defined() on config options.
    - gt powergating work (Riana)
    - Suppress missing out ter rpm protection warning (Rodrigo)
    - Fix a vm leak  (Dafna)
    - Clean up and update 'has_flat_ccs' handling (Lucas)
    - Fix arg to pci_iomap (Lucas)
    - Mark reserved engines in shapshot (Lucas)
    - Don't keep stale pointer (Michal)
    - Fix build warning with CONFIG_PM=n (Arnd)
    - Add a xe_bo subtest for shrinking / swapping (Thomas)
    - Add a warkaround (Tejas)
    - Some display PM work (Maarten)
    - Enable Xe2 + PES disaggregation (Ashutosh)
    - Large xe_mmio rework / cleanup (Matt Roper)
    - A couple of fixes / cleanups in the xe client code (Matt Auld)
    - Fix page-fault handling on closed VMs  (Matt Brost)
    - Fix overflow in OA batch buffer (José)
    - Style fixes (Lucas, Jiapeng, Nitin)
    - Fixes and new development around SRIOV (Michal)
    - Use devm_add_action_or_reset() in gt code (He)
    - Fix CCS offset calculation (Matt Auld)
    - Remove i915_drv.h include (Rodrigo)
    - Restore PCI state on resume (Rodrigo)
    - Fix DSB buffer coherency / Revert DSB disabling (Maarten / Animesh)
    - Convert USM lock to rwsem (Matt Brost)
    - Defer gt-mmio intialization (Matt Roper)
    - meemirq changes (Ilia)
    - Move some PVC related code out of xe-for-CI and to the driver (Rodrigo / Jani)
    - Use a helper for ASID->VM lookup (Matt Brost)
    - Add new PCI id for ARL (Dnyaneshwar)
    - Use Xe2_LPM steering tables for Xe2_HPM (Gustavo)
    - Performance tuning work for media GT and L3 cache flushing (Gustavo)
    - Clean up VM- and exec queue file lock usage (Matt Brost)
    - GuC locking fix (Matt Auld)
    - Fix UAF around queue destruction (Matt Auld)
    - Move IRQ-related registers to dedicated header (Matt Roper)
    - Resume TDR after GT reset (Matt Brost)
    - Move xa_alloc to prevent UAF (Matt Auld)
    - Fix OA stream close (José)
    - Remove unused i915_gpu_error.h (Jani)
    - Prevent null pointer access in xe_migrate_copy (Zhanjun)
    - Fix memory leak when aborting binds (Matt Brost)
    - Prevent UAF in send_recv() (Matt Auld)
    - Fix xa_store() error checking (Matt Auld)
    - drop irq disabling around xa_erase in guc code (Matt Auld)
    - Use fault injection infrastructure to find issues as probe time (Francois)
    - Fix a workaround implementation. (Vinay)
    - Mark wedged_mode debugfs writable (Matt Roper)
    - Fix for prviewous memirq work (Michal)
    - More SRIOV work (Michal)
    - Devcoredump work (John)
    - GuC logging + devcoredump support (John)
    - Don't report L3 bank availability on PTL (Shekhar)
    - Replicate Xe2 PAT settings on Xe2 (Matt Roper)
    - Define Xe3 feature flags (Haridhar)
    - Reuse Xe2 MOCS table on on PTL (Haridhar)
    - Add PTL platform definition (Haridhar)
    - Add MCR steering for Xe3 (Matt)
    - More work around GuC capture for devcoredump (Zhanjun)
    - Improve cache flushing behaviour on bmg (Matt Auld)
    - Fix shrinker test compiler warnings on 32-bit (Thomas)
    - Initial set of workarounds for Xe3 (Gustavo)
    - Extend workaround for xe2lpg (Aradhya)
    - Fix unbalanced rpm put x 2 (Matt Auld)
    
    Signed-off-by: Dave Airlie <airlied@redhat.com>
    
    # -----BEGIN PGP SIGNATURE-----
    #
    # iHUEABYKAB0WIQRskUM7w1oG5rx2IZO4FpNVCsYGvwUCZwekBwAKCRC4FpNVCsYG
    # v32oAQDnIKVwjZecI1V3oUsy2ZE3TKWx8HH4FweT6S5L6tqZwQD/b0vkeA3UaojO
    # 5FIkPEqyHFbrj+Sw7bLonLb3LHv4WAE=
    # =FtY6
    # -----END PGP SIGNATURE-----
    # gpg: Signature made Thu 10 Oct 2024 19:53:11 AEST
    # gpg:                using EDDSA key 6C91433BC35A06E6BC762193B81693550AC606BF
    # gpg: Can't check signature: No public key
    
    # Conflicts:
    #       drivers/gpu/drm/xe/xe_gt_mcr.c
    #       drivers/gpu/drm/xe/xe_tuning.c
    From: Thomas Hellstrom <thomas.hellstrom@linux.intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/Zwekwrak12c5SSgo@fedora

diff --cc drivers/gpu/drm/xe/regs/xe_gt_regs.h
index ac9c437e103d,42dc55cb23f4..d428d04164d9
--- a/drivers/gpu/drm/xe/regs/xe_gt_regs.h
+++ b/drivers/gpu/drm/xe/regs/xe_gt_regs.h
@@@ -393,14 -404,11 +404,17 @@@
  
  #define XE2_GLOBAL_INVAL			XE_REG(0xb404)
  
- #define SCRATCH1LPFC				XE_REG(0xb474)
- #define   EN_L3_RW_CCS_CACHE_FLUSH		REG_BIT(0)
+ #define XE2LPM_L3SQCREG2			XE_REG_MCR(0xb604)
+ 
+ #define XE2LPM_L3SQCREG3			XE_REG_MCR(0xb608)
+ 
+ #define XE2LPM_SCRATCH3_LBCF			XE_REG_MCR(0xb654)
 +
 +#define XE2LPM_L3SQCREG2			XE_REG_MCR(0xb604)
 +
 +#define XE2LPM_L3SQCREG3			XE_REG_MCR(0xb608)
 +
 +#define XE2LPM_SCRATCH3_LBCF			XE_REG_MCR(0xb654)
  
  #define XE2LPM_L3SQCREG5			XE_REG_MCR(0xb658)